crates/bevy_winit/src/lib.rs

```diff
@@ -730,105 +744,9 @@ pub fn winit_runner(mut app: App) {
-                        match config.update_mode(focused) {
-                            UpdateMode::Continuous => {
-                                event_loop.set_control_flow(ControlFlow::Poll);
-                            }
-                            UpdateMode::Reactive { wait }
-                            | UpdateMode::ReactiveLowPower { wait } => {
-                                if let Some(next) = runner_state.last_update.checked_add(*wait) {
-                                    runner_state.scheduled_update = Some(next);
-                                    event_loop.set_control_flow(ControlFlow::WaitUntil(next));
-                                } else {
-                                    runner_state.scheduled_update = None;
-                                    event_loop.set_control_flow(ControlFlow::Wait);
-                                }
-                            }
-                        }
-
-                        if let Some(app_redraw_events) =
-                            app.world.get_resource::<Events<RequestRedraw>>()
-                        {
-                            if redraw_event_reader.read(app_redraw_events).last().is_some() {
-                                runner_state.redraw_requested = true;
-                                event_loop.set_control_flow(ControlFlow::Poll);
-                            }
-                        }
-
-                        if let Some(app_exit_events) = app.world.get_resource::<Events<AppExit>>() {
-                            if app_exit_event_reader.read(app_exit_events).last().is_some() {
-                                event_loop.exit();
-                            }
-                        }
-                    }
-
-                    // create any new windows
-                    // (even if app did not update, some may have been created by plugin setup)
-                    let (
-                        commands,
-                        mut windows,
-                        event_writer,
-                        winit_windows,
-                        adapters,
-                        handlers,
-                        accessibility_requested,
-                    ) = create_window_system_state.get_mut(&mut app.world);
-
-                    create_windows(
-                        event_loop,
-                        commands,
-                        windows.iter_mut(),
-                        event_writer,
-                        winit_windows,
-                        adapters,
-                        handlers,
-                        accessibility_requested,
-                    );
-
-                    create_window_system_state.apply(&mut app.world);
+                let (_, winit_windows, _, _) = event_writer_system_state.get_mut(&mut app.world);
+                for window in winit_windows.windows.values() {
+                    window.request_redraw();
```

Comment:
Preferably also call `Window::request_redraw()` in `WindowEvent::RedrawRequested`.

I would generally recommend not to use `Event::AboutToWait` at all unless you have a very specific use-case in mind.